flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcRowDataLookupFunction.java

```diff
@@ -92,9 +104,18 @@ public JdbcRowDataLookupFunction(
                                 })
                         .toArray(DataType[]::new);
         this.maxRetryTimes = maxRetryTimes;
-        this.query =
+
+        final String baseSelectStatement =
                 options.getDialect()
                         .getSelectFromStatement(options.getTableName(), fieldNames, keyNames);
+        if (conditions == null || conditions.length == 0) {
+            this.query = baseSelectStatement;
+        } else {
+            this.query =
+                    baseSelectStatement
+                            + " AND "
```

Comment:
Why do we always chain with `AND`? is baseSelectStatement guaranteed to have a condition?

Reply:
> Why do we always chain with `AND`? is baseSelectStatement guaranteed to have a condition?

Yes for. lookup join we need at least one lookup key so the baseSelectStatement should have these.